sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml

```diff
@@ -0,0 +1,204 @@
+            app.datasource.cardholder.password=P@ssw0rd
+            app.datasource.cardholder.driverClassName=com.mysql.cj.jdbc.Driver
+          containsSecret: true
+          match: P@ssw0rd
+
+    - id: java-datasource-quoted
+      metadata:
+        name: Database passwords should not be disclosed
+      detection:
+        pre:
+          include:
+            content:
+              - datasource
+        matching:
+          # Same as the file java-datasource-unquoted, except that the value is surrounded by double quotes.
+          # This starts with a negative lookbehind to see if there's a double quote shortly before the string. This is
+          # to prevent false negative matches with code like:
+          #   param += " --spring.datasource.url=" + url
+          #         + " --spring.datasource.username=" + username
+          #         + " --spring.datasource.password=" + password;
+          pattern: "(?<!\"[^\"\\r\\n]{0,6})\\b[0-9a-z_\\-.]+\\.datasource(?:\\.[0-9a-z_\\-.]+)*\\.password[ \\t]*=[ \\t]*\"((?:\\\\\"|[^\"])+?)\""
+          context:
+            - patternAround: "\\b[0-9a-z_\\-.]+\\.datasource(?:\\.[0-9a-z_\\-.]+)*\\.username[ \\t]*="
+        post:
+          # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
+          patternNot: "^(?i)pass(word|wd)?(?-i)$|^[Xx]{3,}$|^%s$|^\\$[A-Z_]+$|^\\$[{(]"
+      examples:
+        - text: >
+            spring.datasource.url="jdbc:mariadb://$NEW_DB_HOST:$NEW_DB_PORT/$NEW_DB_NAME"
+            spring.datasource.username="$NEW_DB_USER"
+            spring.datasource.password="$NEW_DB_PASS"
+          containsSecret: false
+        - text: >
+            spring.datasource.url="jdbc:mysql://localhost:3306/mydatabase"
+            spring.datasource.username="root"
+            spring.datasource.password="password"
+            spring.datasource.driver-class-name="com.mysql.jdbc.Driver"
+          containsSecret: false
+        - text: >
+            param += " --spring.datasource.username=" + username + " --spring.datasource.password=" + password + " --spring.datasource.url=" + url;
+          containsSecret: false
+        - text: >
+            db.default.datasource.username="root"
+            db.default.datasource.password="P@ssw0rd"
+          containsSecret: true
+          match: P@ssw0rd
+
+    - id: jdbc-connection-string
+      metadata:
+        name: Database passwords should not be disclosed
+      detection:
+        pre:
+          include:
+            content:
+              - jdbc
+          reject:
+            ext:
+              - .adoc
+              - .example
+              - .html
+              - .md
+              - .mdx
+              - .template
+        matching:
+          # JDBC URIs are in the form "jdbc:protocol://server:port/database?user=...&password=...&paramN=..."
+          pattern: "\\bjdbc:[^\"\\s?]+\\?(?:[^&\"\\s]+&)*password=([^&\"\\s]+)(?:[&\"\\s])"
```

Comment:
I feel like this pattern will overlap the mysql one in a lot of cases. Other than that LGTM

Reply:
There shouldn't be overlap because the other URI-like patterns expect `protocol://username:password@host:port` whereas this requires the username and password to be passed as query parameters.